src/openms/source/ANALYSIS/OPENSWATH/TargetedSpectraExtractor.cpp

```diff
@@ -0,0 +1,581 @@
+    params.setValidStrings("mz_unit_is_Da", ListUtils::create<String>("false,true"));
+
+    params.setValue("sgolay_frame_length", 15, "The number of subsequent data points used for smoothing.\nThis number has to be uneven. If it is not, 1 will be added.");
+    params.setValue("sgolay_polynomial_order", 3, "Order of the polynomial that is fitted.");
+    params.setValue("gauss_width", 0.2, "Gaussian width in Da or ppm, estimated peak size.");
+    params.setValue("use_gauss", "true", "Use Gaussian filter for smoothing (alternative is Savitzky-Golay filter)");
+    params.setValidStrings("use_gauss", ListUtils::create<String>("false,true"));
+    params.setValue("signal_to_noise", 1.0, "Signal-to-noise threshold at which a peak will not be extended any more. Note that setting this too high (e.g. 1.0) can lead to peaks whose flanks are not fully captured.");
+    params.setMinFloat("signal_to_noise", 0.0);
+
+    params.setValue("peak_height_min", 0.0, "Minimum picked peak's height.");
+    params.setMinFloat("peak_height_min", 0.0);
+    params.setValue("peak_height_max", 4e6, "Maximum picked peak's height.");
+    params.setMinFloat("peak_height_max", 0.0);
+    params.setValue("fwhm_threshold", 0.0, "Picked peak's FWHM threshold.");
+    params.setMinFloat("fwhm_threshold", 0.0);
+
+    params.setValue("tic_weight", 1.0, "TIC weight when scoring spectra.");
+    params.setMinFloat("tic_weight", 0.0);
+    params.setValue("fwhm_weight", 1.0, "FWHM weight when scoring spectra.");
+    params.setMinFloat("fwhm_weight", 0.0);
+    params.setValue("snr_weight", 1.0, "SNR weight when scoring spectra.");
+    params.setMinFloat("snr_weight", 0.0);
+  }
+
+  void TargetedSpectraExtractor::annotateSpectra(
+    const std::vector<MSSpectrum>& spectra,
+    const TargetedExperiment& targeted_exp,
+    std::vector<MSSpectrum>& annotated_spectra,
+    FeatureMap& features
+  )
+  {
+    annotated_spectra.clear();
+    features.clear(true);
+    const std::vector<ReactionMonitoringTransition> transitions = targeted_exp.getTransitions();
+    for (UInt i=0; i<spectra.size(); ++i)
+    {
+      MSSpectrum spectrum = spectra[i];
+      // It is supposed to have RT in minutes in target list file, therefore we divide by 60.0
+      const double spectrum_rt = spectrum.getRT() / 60.0;
+      const double rt_left_lim = spectrum_rt - getRTWindow() / 60.0 / 2.0;
+      const double rt_right_lim = spectrum_rt + getRTWindow() / 60.0 / 2.0;
+      const std::vector<Precursor> precursors = spectrum.getPrecursors();
+      if (precursors.size() < 1)
+      {
+        throw Exception::IllegalArgument(__FILE__, __LINE__, OPENMS_PRETTY_FUNCTION,
+                                         "Spectrum does not contain precursor info.");
+      }
+      const double spectrum_mz = precursors[0].getMZ();
+      const double mz_left_lim = spectrum_mz - getMZTolerance();
+      const double mz_right_lim = spectrum_mz + getMZTolerance();
```

Comment:
why is the RT in a window and needs to be divided by 2.0 and the m/z not? I would prefer consistency

Reply:
PS: it may also be useful to express m/z tolerance in ppm

Reply:
RT is a window (hence divided by two) while m/z is a tolerance (hence +/-).